web3/middleware/signing.py

```diff
@@ -0,0 +1,92 @@
+from eth_account.local import (
+    LocalAccount,
+)
+from eth_keys.datatypes import (
+    PrivateKey,
+)
+from eth_utils import (
+    is_same_address,
+)
+from toolz import (
+    compose,
+)
+
+from web3.utils.formatters import (
+    apply_formatter_if,
+)
+from web3.utils.transactions import (
+    fill_nonce,
+    fill_transaction_defaults,
+)
+
+to_hexstr_from_eth_key = lambda x: x.to_hex()  # noqa: E731
+
+
+def is_eth_key(value):
+    if isinstance(value, PrivateKey):
+        return True
+    return False
+
+
+def raw_key_formatter(private_key):
+    formatter = compose(
+        apply_formatter_if(is_eth_key, to_hexstr_from_eth_key),
+    )
+    return formatter(private_key)
+
+
+def to_account(web3, private_key):
+
+    if isinstance(private_key, LocalAccount):
+        return private_key
+
+    elif isinstance(private_key, (PrivateKey, str, bytes,)):
```

Comment:
The acceptance of both `str` and `bytes` seems off here, or at least like they should require different logic (unless maybe `web3.eth.account.privateKeyToAccount` gracefully handles hex encoded private keys in which case ignore I said anything).

Reply:
Yup, `privateKeyToAccount` assumes hex-encoding if a `str`: http://eth-account.readthedocs.io/en/latest/eth_account.html#eth_account.account.Account.privateKeyToAccount